flask_server.py

```diff
@@ -85,12 +84,18 @@ def settings_page():
             if setting_name == "extra_ids":
                 extra_ids = request.form["extra_ids"].splitlines()
                 settings.update("extra_ids", extra_ids)
+            elif setting_name in ["loop_delay", "loop_variation"] and len(request.form[setting_name]) != 0:
+                settings.update(setting_name, int(request.form[setting_name]))
             elif len(request.form[setting_name]) != 0:  # Updates other non-null settings.
                 settings.update(setting_name, request.form[setting_name])
+        loop_args = {"loop_delay": settings.get("loop_delay"), "loop_variation": settings.get("loop_variation")}
+        scraper_runner.updateDelay(**loop_args)
```

Comment:
Move this out of the loop as you always need it, this way you don't request it two times with settings.get